mlabs/src/Mlabs/Lending/Contract/Server.hs

```diff
@@ -230,3 +234,27 @@ findInputStateData lid = do
   maybe err pure $ firstJust readDatum' txOuts
   where
     err = Contract.throwError $ StateMachine.toLendexError "Can not find Lending app instance"
+
+{- | Find datum, is not the best way to find the datum. The logic is not neccesarily correct
+ as many datum can be sitting at App address, and there is no way of enforcing which one is
+ correct.
+ todo: review logic
```

Comment:
@cstml change this to -- "add a unique NFT to distinguish between utxos / review logic"